connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/PartyConnectorImpl.java

```diff
@@ -323,6 +311,47 @@ public void createUsers(String institutionId, String productId, String userId, C
     }
 
 
+    @Override
+    public void checkExistingRelationshipRoles(String institutionId, String productId, CreateUserDto userDto, String userId) {
+        log.trace("checkExistingRelationshipRoles start");
+        log.debug(LogUtils.CONFIDENTIAL_MARKER, "checkExistingRelationshipRoles institutionId = {}, productId = {}, createUserDto = {}, userId = {}", institutionId, productId, userDto, userId);
+
+        Map<PartyRole, List<User>> partyRoleToUsersMap = getPartyRoleListMap(userId, userDto);
+
+        UserInfo.UserInfoFilter userInfoFilter = new UserInfo.UserInfoFilter();
+        userInfoFilter.setProductId(Optional.of(productId));
+        userInfoFilter.setUserId(Optional.ofNullable(userId));
+        userInfoFilter.setAllowedState(Optional.of(EnumSet.of(ACTIVE, SUSPENDED)));
+
+        RelationshipsResponse institutionRelationships = partyProcessRestClient.getUserInstitutionRelationships(institutionId, EnumSet.allOf(PartyRole.class), userInfoFilter.getAllowedStates().orElse(null), userInfoFilter.getProductId().map(Set::of).orElse(null), userInfoFilter.getProductRoles().orElse(null), userInfoFilter.getUserId().orElse(null));
+        if (!institutionRelationships.isEmpty()) {
+            Set<PartyRole> roles = partyRoleToUsersMap.keySet();
+            List<PartyRole> partyRoles = institutionRelationships.stream().map(RelationshipInfo::getRole).collect(Collectors.toList());
+
+            if (!roles.contains(PartyRole.OPERATOR) || !(partyRoles.contains(PartyRole.OPERATOR))) {
```

Comment:
Are you searching only for Operator Role?

Reply:
The only accepted case is the multi-role Operator on the Interoperabilità product. Therefore, an exception is thrown when the role entered on the dashboard is not operator or when the role retrieved from party is not operator, according to the truth table of a logical OR port (where the 0-0 case is the only accepted case).